brjs-core/src/main/java/org/bladerunnerjs/model/engine/AbstractRootNode.java

```diff
@@ -89,7 +89,7 @@ public void registerNode(Node node) throws NodeAlreadyRegisteredException
 	public void clearRegisteredNode(Node node) {
 		String normalizedPath = node.dir().getAbsolutePath();
 		List<Node> nodesForPath = nodeCache.get(normalizedPath);
-		nodesForPath.remove(node.getTypeName());
+		nodesForPath.remove(node);
```

Comment:
That method is not covered by a test + i can't find anywhere where its being used, so maybe it can be removed? But, List remove method takes an object, not a string, so applying the fix won't do any harm, so we should either apply the fix, or delete the method